gradle-plugin-testing/src/test/groovy/com/palantir/gradle/plugintesting/TestDependencyVersionsTaskSpec.groovy

```diff
@@ -0,0 +1,118 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.plugintesting
+
+import static TestDependencyVersions.resolve
+
+class TestDependencyVersionsTaskSpec extends AbstractTestingPluginSpec {
+
+    File outputFile
+
+    def setup() {
+        outputFile = new File(projectDir,'build/plugin-testing/dependency-versions.properties')
+    }
+
+    def 'write versions without GCV'() {
+        given:
+        buildFile.text = basicBuildFile
+
+        when:
+        def result = runTasksSuccessfully('writeTestDependencyVersions')
+
+        then:
+        outputFile.exists()
+        !outputFile.text.contains('null')
+        outputFile.text.contains('com.google.guava:guava=33.3.1-jre')
+        outputFile.text.contains('org.junit.jupiter:junit-jupiter=5.11.3')
+        outputFile.text.contains('com.netflix.nebula:nebula-test=10.6.1')
+        outputFile.text.contains('com.palantir.gradle.consistentversions:gradle-consistent-versions=2.31.0')
+        outputFile.text.contains('com.palantir.gradle.plugintesting:plugin-testing-core')
+    }
+
+    def 'write versions with GCV'() {
+        given:
+        writeDependenciesVersionsFile()
+        buildFile << buildFileWithGcv
+        TestContentHelpers.addVersionsToPropsFile(file('versions.props'), ['org.junit.jupiter:junit-jupiter', 'com.netflix.nebula:nebula-test', 'com.google.guava:guava', 'com.palantir.gradle.consistentversions:gradle-consistent-versions'])
+        runTasksSuccessfully('writeVersionLocks')
+
+        when:
+        def result = runTasksSuccessfully('writeTestDependencyVersions')
+
+        then:
+        outputFile.exists()
+        !outputFile.text.contains('null')
+        outputFile.text.contains('org.junit.jupiter:junit-jupiter')
```

Comment:
no because the versions will change as the versions within the current project change.  So we could do something else to lookup the versions (i.e. parse the lock file) but that seems a bit strange

Reply:
I couldn't remove the use of TestContentHelpers.addVersionsToPropsFile and just hard-code the set of strings to write into versions.props to make it more clear what we are testing.